File: backend/transactions/device_models/create_device_model_transaction.py
from device_models.add_device_model import add_device_model
from schemas.device_model_schemas import AddDeviceModelRequest
from supabase import Client

def create_device_model_transaction(
                                        client:Client,
                                        device_model: AddDeviceModelRequest,
                                        hospital_id: str
                                    ):
    print("create_device_model_transaction")

    return add_device_model(
                                client=client, 
                                device_model=device_model,
                                hospital_id=hospital_id
                            )

File: frontend/app/api/transactions/deviceModels/createDeviceModelTransaction.ts
import { API_BASE_URL } from "../../client/apiClient"
import { CreateDeviceModelFrontType } from "../../../types/deviceModelTypes"
import { getDeviceModelsFromApi } from "../../deviceModels/fetchDeviceModels"
import {
         normalizeDeviceModel,
         toCreateDeviceModelRequest
       } from "../../../utils/deviceModelMapper"
import { authFetch } from "../../client/apiClient"

type CreateDeviceModelTransactionParams = {
                                             deviceModel: CreateDeviceModelFrontType
                                             setDeviceModels: any
                                             onClose?: () => void
                                           }
export async function createDeviceModelTransaction({
                                                     deviceModel,
                                                     setDeviceModels,
                                                     onClose
                                                   }: CreateDeviceModelTransactionParams
                                                 )
{
  console.log("createDeviceModelTransaction")
  await authFetch(
                `${API_BASE_URL}/device-models`,
                {
                  method: "POST",
                  headers: {
                              "Content-Type":
                              "application/json"
                            },
                  body: JSON.stringify(toCreateDeviceModelRequest(deviceModel))
                }
              )
  const deviceModels =await getDeviceModelsFromApi()
  setDeviceModels(deviceModels.map(normalizeDeviceModel))
  if (onClose) {onClose()}
}